### ISSUE TITLE
Rotate transition not working for OSX

### ISSUE THREAD

**Issue by IanMayo** [debrief/limpet#265]
@theanuradha is developing under OSX.

But, the Nebula Transition Widget isn't working for OSX.

It is working for Win 7, Win 8, and Ubuntu 15.10

There is value in a brief investigation into the problem, with an initial budget of 2 hours.  If a fix isn't found, then the issue will be deferred for the future.
## Update

This is an existing Nebula transitions issue:
https://bugs.eclipse.org/bugs/show_bug.cgi?id=404842

**Comment by IanMayo** [comment]
@theanuradha - I've had a freelancer looking at the issue.  He has come up with this fix:
https://github.com/cantupolo/nebula/commit/a2b5103fbd996b323e041bc6b227915182c75393

Would you mind checking if that fixes your osx rotate transition?

The freelancer also said this:

```
Ok...well, if you don't succeed to see the left/right transitions, please try to decrease the transitions quality on the component's demo. I can see the left/right transitions if I decrease the quality to 97%.
```

cheers,
Ian

**Comment by theanuradha** [comment]
@IanMayo  sure I will

**Comment by theanuradha** [comment]
@IanMayo  we got slight flick on rotation but yes it works now.

**Comment by theanuradha** [comment]
@IanMayo  left & right also need to reduce quality to work

**Comment by theanuradha** [comment]
patch in action 
http://sendvid.com/1qi7dsj9

**Comment by IanMayo** [comment]
Ok, @theanuradha  could you produce a PR to trial the new code in Limpet?  Can we override the Nebula default behaviour in our code, so we don't need a new .jar?

My suspicion is that we probably can't , since so many SWT controls say "this is not meant to be subclassed".

**Comment by theanuradha** [comment]
https://github.com/debrief/limpet/pull/354

**Comment by IanMayo** [comment]
@theanuradha - had another update from Rodrigo:

# 

I commited the fix update to GitHub, at the same location: https://github.com/cantupolo/nebula/tree/stw-macosx-fix/widgets/effects/org.eclipse.nebula.effects.stw/src/org/eclipse/nebula/effects/stw, stw-macosx-fix branch.
I tested the fix on MacOS X and Windows Vista. It worked with no problems.
The fix was about using data buffers before drawing the transition images with SWT's gc.drawImage() method, but this is done only for MacOSX. I maintained the gc.drawImage() calls for Linux and Windows.
I still have to test this on Linux. You can start testing what I've commited, or wait for me to test this on Linux, if you want.

# 

@theanuradha : Could you trial this fix using Rodrigo's demo app in OSX,  and then in Limpet?

**Comment by theanuradha** [comment]
@IanMayo  IM on it

**Comment by theanuradha** [comment]
@IanMayo  I have pushed new patched changes and it is lot smooth on mac/osx. WIll test windows now

**Comment by theanuradha** [comment]
https://github.com/debrief/limpet/pull/354 changes are in

**Comment by theanuradha** [comment]
@IanMayo  tested on windows works smoothly.  only pending issue is https://github.com/debrief/limpet/issues/337 but this is more on our code I will work on it separately

**Comment by IanMayo** [comment]
that's great. will review tomorrow

**Comment by theanuradha** [comment]
@IanMayo I just push fix for #337 [Flicker on Stacked Charts rotate transition] part of this PR #354

**Comment by IanMayo** [comment]
Done, Rodrigo has fixed this.
